src/sentry/api/paginator.py

```diff
@@ -512,27 +550,33 @@ def _is_asc(self, is_prev):
 
     def _build_combined_querysets(self, value, is_prev, limit, extra):
         asc = self._is_asc(is_prev)
+        combined_querysets = list()
+        for queryset in self.querysets:
+            item = queryset.first()
```

Comment:
We probably don't need to do this query if we inspect the querysets and just determine model instead.